packages/js/email-editor/src/components/personalization-tags/category-section.tsx

```diff
@@ -9,6 +10,13 @@ const CategorySection = ( {
 	groupedTags: Record< string, PersonalizationTag[] >;
 	activeCategory: string | null;
 } ) => {
+	const { togglePersonalizationTagsModal } = useDispatch( storeName );
+
+	const { onInsert } = useSelect(
+		( select ) => select( storeName ).getPersonalizationTagsState(),
+		[]
+	);
```

Comment:
Can you please elaborate on why we need to add the callback into the state? It looks suspicious to me. I would expect we would use a custom action, perhaps a hook.